pkg/cmd/pulumi/convert.go

```diff
@@ -497,13 +537,153 @@ func runConvert(
 	return nil
 }
 
-func newPluginContext(cwd string) (*plugin.Context, error) {
-	sink := diag.DefaultSink(os.Stderr, os.Stderr, diag.FormatOptions{
-		Color: cmdutil.GetGlobalColorization(),
-	})
-	pluginCtx, err := plugin.NewContext(sink, sink, nil, nil, cwd, nil, true, nil)
+// getPackagesToGenerateSdks parses the pcl files back in to read the package
+// blocks and for sdk generation.
+func getPackagesToGenerateSdks(
+	sourceDirectory string,
+) (map[string]*schema.PackageDescriptor, hcl.Diagnostics, error) {
+	// TODO use the package block to get the packages and parameterize them and generate SDKs.
+	files, err := os.ReadDir(sourceDirectory)
 	if err != nil {
-		return nil, err
+		return nil, nil, fmt.Errorf("could not read sourceDirectory: %w", err)
+	}
+
+	parser := hclsyntax.NewParser()
+	_, err = pcl.ParseFiles(parser, sourceDirectory, files)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not parse pcl file: %w", err)
+	}
+
+	allPackageDescriptors := make(map[string]*schema.PackageDescriptor)
+
+	var diagnostics hcl.Diagnostics
+	for _, file := range parser.Files {
+		packageDescriptors, diags := pcl.ReadPackageDescriptors(file)
+		diagnostics = append(diagnostics, diags...)
+		for packageName, descriptor := range packageDescriptors {
+			if _, ok := allPackageDescriptors[packageName]; ok {
+				diagnostics = append(diagnostics, errorf(file.Body.Range(), "package %q was already defined", packageName))
+				continue
+			}
+			allPackageDescriptors[packageName] = descriptor
+		}
+	}
+
+	if len(diagnostics) != 0 {
+		var errorDiags hcl.Diagnostics
+		for _, d := range diagnostics {
+			if d.Severity == hcl.DiagError {
+				errorDiags = append(errorDiags, d)
+			}
+		}
+
+		if len(errorDiags) != 0 {
+			return nil, diagnostics, nil
+		}
+	}
+
+	return allPackageDescriptors, nil, nil
+}
+
+func generateAndLinkSdksForPackages(
+	ctx context.Context,
+	ws pkgWorkspace.Context,
+	language string,
+	sdkTargetDirectory string,
+	convertOutputDirectory string,
+	pkgs map[string]*schema.PackageDescriptor,
+) error {
+	for _, pkg := range pkgs {
+		tempOut, err := os.MkdirTemp("", "gen-sdk-for-dependency-")
+		if err != nil {
+			return fmt.Errorf("failed to create temporary directory: %w", err)
+		}
+
+		var parameterizationValue []byte
+		if pkg.Parameterization != nil {
+			parameterizationValue = pkg.Parameterization.Value
+		} else {
+			// Only generate SDKs for packages that have parameterization for now, others should be implicit.
+			continue
+		}
+
+		pkgSchema, err := pcmd.SchemaFromSchemaSourceValueArgs(ctx, pkg.Name, parameterizationValue)
+		if err != nil {
+			return fmt.Errorf("error creating pckage schema: %w", err)
```

Comment:
```suggestion
			return fmt.Errorf("creating package schema: %w", err)
```
Typo but also this applies to a lot of the error messages in this file, do not prefix them all with error: https://preslav.me/2023/04/14/golang-error-handling-is-a-form-of-storytelling/#crafting-a-good-message-is-key